### ISSUE TITLE
Checking the parallel status from within the function

### ISSUE THREAD

**Issue by alyashenko** [joblib/joblib#1330]
Hello, is there a way to determine inside the called parallel function that it is running in parallel and not just single run from the console? In the multiprocessing pipeline this can be achieved by checking value `current_process().daemon` - is there a simple counterpart to this call in joblib?

**Comment by ogrisel** [comment]
I don't think we have a dedicated API for that.

For process based parallelism, you can look at whether the pid (`os.getpid()`) is the same in the function as the one measured in the main process.

**Comment by alyashenko** [comment]
Yes, I have done so so far, but it won't work in the case of thread-based parallelism. Anyway, thanks for your feedback.

**Comment by tomMoral** [comment]
The issue with thread/process id is that such a check would be backend-specific.
Could you share your use case for such a feature?

A potential candidate for testing this is would be to rely on the [`ParallelBackend.nesting_level`](https://github.com/joblib/joblib/blob/main/joblib/_parallel_backends.py#L44) which we use to prevent nested calls in joblib.

**Comment by alyashenko** [comment]
Hi @tomMoral, specifically my problem is to optimize some complex objective function by grid search of its parameters. When parallel search is run under joblib - it is possible to perform calculations inside the objective function not completely, returning only one number as a metric. When the same is run in single mode - more detailed results are calculated and returned. That's why I'm need to know the mode of launching inside it.
